setup.sh is missing bats-core installation

`setup.sh` installs `markdownlint-cli2` but does not install `bats-core`, which is required to run `make test`. After running `./setup.sh`, `bats` is not on PATH and `make test` fails immediately with:

```
make: bats: No such file or directory
make: *** [test] Error 1
```

`setup.sh` should install both dev dependencies so a contributor can run the full `make all` (lint + test) after a single setup step.

## Steps to Reproduce

1. Fresh environment with Homebrew but no `bats-core`
2. Run `./setup.sh`
3. Run `make test`
4. Observe `bats: No such file or directory`

## Expected Behavior

After `./setup.sh`, both `make lint` and `make test` succeed without requiring manual installation of additional tools.

## Proposed Fix

Add `brew install bats-core` to `setup.sh` after the `markdownlint-cli2` install, and update the success message to mention both tools.

--- a/setup.sh
+++ b/setup.sh
@@ -24,9 +24,15 @@ brew install markdownlint-cli2
 
 echo "✓ markdownlint-cli2 installed: $(markdownlint-cli2 --version)"
 
+# Install bats-core
+echo "📦 Installing bats-core..."
+brew install bats-core
+
+echo "✓ bats-core installed: $(bats --version)"
+
 echo ""
 echo "✅ Setup complete!"
 echo ""
 echo "Next steps:"
-echo "  Run make lint before pushing a PR"
+echo "  Run make lint && make test before pushing a PR"
 echo ""

